packages/kirby/src/views/Web3Enable.tsx

```diff
@@ -8,32 +11,60 @@ import {
   CenteredPage,
 } from "@kirby-web3/child-react";
 import { EthereumChildPlugin } from "@kirby-web3/plugin-ethereum";
-import { RouteComponentProps } from "@reach/router";
 import { ViewPlugin } from "@kirby-web3/child-core";
+import { ClipLoader } from "@joincivil/elements";
 
 interface Web3EnableProps extends RouteComponentProps {
   network?: string;
 }
 
 export const Web3Enable: React.FC<Web3EnableProps> = ({ network }) => {
   const ctx = React.useContext(CoreContext);
+  const [status, setStatus] = React.useState("provided");
+
   const requestID = useSelector((state: any) => {
     if (state.view && state.view.queue && state.view.queue[0]) {
       return state.view.queue[0].requestID;
     }
   });
-  function selection(provider: string): void {
+
+  React.useEffect(() => {
+    const queryParams = queryString.parse(window.location.search);
+    if (queryParams.providerPreference) {
+      selection(queryParams.providerPreference as string).catch(err => console.log("error with selection", err));
+    } else {
+      setStatus("select");
+    }
+  }, []);
+
+  async function selection(provider: string): Promise<void> {
+    const ethPlugin = ctx.core.plugins.ethereum as EthereumChildPlugin;
     console.log("selected:", provider);
-    (ctx.core.plugins.ethereum as EthereumChildPlugin).enableWeb3(requestID, provider, network as any).catch(err => {
+    try {
+      setStatus("enabling provider");
+      await ethPlugin.enableWeb3(requestID, provider, network as any);
+      setStatus("done");
+      (ctx.core.plugins.view as ViewPlugin).completeView();
+    } catch (err) {
       console.log("error with enableWeb3: ", err);
-    });
-    (ctx.core.plugins.view as ViewPlugin).completeView();
+      (ctx.core.plugins.view as ViewPlugin).completeView();
+    }
   }
+
+  if (status === "provided") {
+    return <></>;
+  } else if (status === "enabling provider") {
+    return (
+      <CenteredPage>
+        <ClipLoader />
```

Comment:
can we add some text here like how we have "waiting for wallet connection" and "waiting for signuatre"?

Reply:
good call, added